lib/server/services/run/RunService.js

```diff
@@ -145,6 +146,17 @@ class RunService {
             runType = await getOrCreateRunType({ name: runTypeName });
         }
 
+        if (user_o2_start?.externalId) {
```

Comment:
user_o2_start and stop are numbers no? Where does the externalId comes from?!

Reply:
Because the gRPC run service sends a User message and not an id, because the external AliECS Core does not know the user id in the bookkeeping db, so the gRPC run service sends a User without id, but with an externalId. And with the externalId I made a function getOrCreate to get the user from the bookkeeping db. After doing that function, I have the full user available, now including the id, so that I can use that for further processing (the run model can now use this user id acquired in the described functionality).

This is described in the parent ticket: "The protofile should be updated to include a new struct defined in common.proto which defines a user containing: number external_id, string name (which is actually a username)."

Reply:
Ok you are right